proxy/handler.go

```diff
@@ -67,6 +67,8 @@ func (s *handler) handler(srv interface{}, serverStream grpc.ServerStream) error
 	fullMethodName := lowLevelServerStream.Method()
 	// We require that the director's returned context inherits from the serverStream.Context().
 	outgoingCtx, backendConn, err := s.director(serverStream.Context(), fullMethodName)
+	defer backendConn.Close()
```

Comment:
This should go through the directory, IMO.  Also this shouldn't occur unless the call succeeds, otherwise you can get a NPE panic since backendConn may be nil.

Need for backend cleanup was noted in #16, which I've implemented on a separate branch: https://github.com/vgough/grpc-proxy/blob/master/proxy/handler.go#L71

Reply:
You're responding to a comment that's over 4 years old.  I'm not using this anymore. Be sure to test your setup under load. Yes, at the very minimum there should be a nil check to avoid a NPE.